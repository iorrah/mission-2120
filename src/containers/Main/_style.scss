.main {
  display: flex;
  align-items: stretch;
}

.main__container {
  flex-grow: 9;
  padding: 2rem;
}

.main__delimiter {
  width: 1024px;
  margin: 0 auto;
}

.main__header {
  display: flex;
  justify-content: space-between;
  margin: 0 0 1rem;
}

.main__title {
  font-size: 2rem;
  line-height: normal;
  font-weight: 500;
}

.main__counter {
  font-size: 1.3rem;
  margin-left: 0.5rem;
}

.main__button-light {
  margin: 0 0.8rem 0 0;
}

.main__filter-icon {
  display: inline-block;
  margin: -0.2rem 0 0 0.5rem;
}

.main__plus-icon {
  display: inline-block;
  margin: -0.2rem 0 0 0.5rem;
}

.main__table {
  width: 100%;
}

.main__table-body {
  background-color: $white;
}

.main__table-header-row > th {
  padding: 0.7rem 0.7rem;
  font-weight: normal;
  font-size: 0.8rem;
  color: gray;
}

.main__table-header-checkbox {
  width: 5%;
}

.main__table-header-id {
  width: 10%;
}

.main__table-header-logo {
  width: 10%;
}

.main__table-header-name {
  width: 18%;
}

.main__table-header-budget {
  width: 15%;
}

.main__table-header-budget-spent {
  width: 17%;
}

.main__table-header-date {
  width: 20%;
}

.main__checkbox {
  font-size: 1.3rem;
}

.main__table-row {
  cursor: pointer;
  border-bottom: 1px solid $iceberg;
}

.main__table-row > td {
  padding: 0.3rem 0.7rem;
  font-size: 0.9rem;
  line-height: normal;
}

.main__row-logo {
  width: 2.3rem;
}

.main__row-action {
  width: 1rem;
  float: right;
}

.modal-content {
  border: 0;
  border-radius: 7px;
  box-shadow: 0 0 20px 40px #0000000d;

  .close {
    font-size: 2rem;
    font-weight: 400;
  }
}

.modal-header {
  padding-top: 2rem;
}

.modal__logo {
  width: 2.3rem;
  margin: 0 0.5rem 0 0;
  position: relative;
  top: -0.2rem;
}

.modal-header,
.modal-footer {
  border-color: $iceberg;
}

.modal__body {
  padding: 1rem 0 0.5rem;
}

.modal__label {
  margin: 7px 0 0;
  text-align: right;
}

.modal__check-icon {
  display: inline-block;
  margin: -0.2rem 0 0 0.5rem;
}

.modal__close-icon {
  display: inline-block;
  margin: -0.1rem 0 0 0.5rem;
  width: 0.9rem;
}
