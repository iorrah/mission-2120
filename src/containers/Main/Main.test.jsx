import React from "react";
import { render } from "@testing-library/react";
import { shallow, mount } from "enzyme";
import Main from "./Main";
import consumersRaw from "../../api/consumers.js";
import consumers from "../../api/consumers.js";

const mockedConsumer = {
  id: 1,
  name: "Starbucks",
  budget: 94726.572,
  budget_spent: 5783.7122,
  date_of_first_purchase: "2119-03-16"
};

const mockedConsumerDraft = {
  ...mockedConsumer,
  budget_untouched: "94.726,57"
};

describe("Main", () => {
  it("should shallow correctly", () => {
    expect(shallow(<Main />)).toMatchSnapshot();
  });

  it("should mount correctly", () => {
    expect(mount(<Main />)).toMatchSnapshot();
  });

  it("should render correctly", () => {
    expect(render(<Main />)).toMatchSnapshot();
  });

  it("should render the `.main__container` div", () => {
    const wrapper = shallow(<Main />);
    expect(wrapper.find(".main__container").exists()).toBeTruthy();
  });

  it("should render the `.main__delimiter` inside `.main__container`", () => {
    const wrapper = shallow(<Main />);
    expect(
      wrapper.find(".main__container .main__delimiter").exists()
    ).toBeTruthy();
  });

  it("should not render the `.modal` on page load", () => {
    const wrapper = shallow(<Main />);
    expect(wrapper.find(".modal").exists()).toBeFalsy();
  });

  it("should render the Filter Consumers list button", () => {
    const wrapper = shallow(<Main />);
    expect(wrapper.find(".main__button-light").exists()).toBeTruthy();
  });

  it("should render the Add Consumer button", () => {
    const wrapper = shallow(<Main />);
    expect(wrapper.find(".main__button-primary").exists()).toBeTruthy();
  });

  it("should render the page `.main__title`", () => {
    const wrapper = shallow(<Main />);
    expect(wrapper.find("h1.main_title").exists).toBeTruthy();
  });

  it("should render the page title properly", () => {
    const { getByText } = render(<Main />);
    const linkElement = getByText(/Consumers/i);
    expect(linkElement).toBeInTheDocument();
  });

  it("should render a counter within `.main__title`", () => {
    const wrapper = shallow(<Main />);
    expect(wrapper.find(".main_title main__counter").exists).toBeTruthy();
  });

  it("should render a counter within `.main__title`", () => {
    const wrapper = shallow(<Main />);
    const counter = wrapper.find(".main__counter").first();
    expect(counter.text()).toBe(`(${consumersRaw.length})`);
  });

  it("should render 8 <th> columns", () => {
    const wrapper = shallow(<Main />);
    expect(wrapper.find("th").length).toBe(8);
  });

  it("should load with default state", () => {
    const wrapper = shallow(<Main />);
    expect(wrapper.state().consumers).toEqual(consumersRaw);
    expect(wrapper.state().modalOpen).toEqual(false);
    expect(wrapper.state().currentConsumer).toEqual({});
    expect(wrapper.state().modalErrorMessage).toEqual("");
  });

  describe("handleOpenModal", () => {
    it("should update state to modal-opened state", () => {
      const wrapper = shallow(<Main />);
      const instance = wrapper.instance();
      expect(wrapper.state("modalOpen")).toBe(false);
      instance.handleOpenModal(mockedConsumer);
      expect(wrapper.state("modalOpen")).toBe(true);

      expect(wrapper.state("currentConsumer")).toEqual(mockedConsumerDraft);
    });
  });

  describe("handleCloseModal", () => {
    it("should update state to modal-closed state", () => {
      const wrapper = shallow(<Main />);
      wrapper.setState({ modalOpen: true });
      const instance = wrapper.instance();
      instance.handleCloseModal();
      expect(wrapper.state("modalOpen")).toBe(false);
      expect(wrapper.state("currentConsumer")).toEqual({});
      expect(wrapper.state("modalErrorMessage")).toEqual("");
    });
  });

  describe("handleInputChange", () => {
    it("should update state with raw new budget", () => {
      const wrapper = shallow(<Main />);

      wrapper.setState({
        modalOpen: true,
        currentConsumer: mockedConsumerDraft
      });

      const instance = wrapper.instance();
      instance.handleInputChange("111.222,33", 111222.33);

      expect(wrapper.state("currentConsumer")).toEqual({
        ...mockedConsumerDraft,
        budget: "111.222,33",
        budget_raw: 111222.33
      });
    });
  });

  describe("handleSaveChanges", () => {
    it("should just close modal if budget didn't change", () => {
      const wrapper = shallow(<Main />);

      const currentConsumer = {
        ...mockedConsumer,
        budget: "94.726,57"
      };

      wrapper.setState({
        modalOpen: true,
        currentConsumer,
        modalErrorMessage: ""
      });

      const instance = wrapper.instance();
      instance.handleSaveChanges();
      const consumers = instance.getConsumers();

      expect(wrapper.state("modalErrorMessage")).toBe("");
      expect(wrapper.state("currentConsumer")).toEqual({});
      expect(wrapper.state("modalOpen")).toEqual(false);
      expect(wrapper.state("consumers")).toEqual(consumers);
    });

    it("should validate non decimal numbers", () => {
      const wrapper = shallow(<Main />);

      const currentConsumer = {
        ...mockedConsumer,
        budget_raw: "abc"
      };

      wrapper.setState({
        modalOpen: true,
        currentConsumer,
        modalErrorMessage: ""
      });

      const instance = wrapper.instance();
      instance.handleSaveChanges();

      expect(wrapper.state("modalErrorMessage")).toBe(
        "Ops... the budget must be a valid number"
      );

      expect(wrapper.state("currentConsumer")).toEqual(currentConsumer);
      expect(wrapper.state("modalOpen")).toEqual(true);
    });

    it("should validate budgets smaller than spent budget", () => {
      const wrapper = shallow(<Main />);

      const currentConsumer = {
        ...mockedConsumer,
        budget_raw: 1
      };

      wrapper.setState({
        modalOpen: true,
        currentConsumer,
        modalErrorMessage: ""
      });

      const instance = wrapper.instance();
      instance.handleSaveChanges();

      expect(wrapper.state("modalErrorMessage")).toBe(
        "Ops... the budget must be higher than the spent budget (â‚¬5.783,71)"
      );

      expect(wrapper.state("currentConsumer")).toEqual(currentConsumer);
      expect(wrapper.state("modalOpen")).toEqual(true);
    });

    it("should save changes when budget is valid", () => {
      const wrapper = shallow(<Main />);

      wrapper.setState({
        modalOpen: true,
        currentConsumer: mockedConsumer,
        modalErrorMessage: ""
      });

      const instance = wrapper.instance();
      instance.handleSaveChanges();

      expect(wrapper.state("modalOpen")).toEqual(false);
      expect(wrapper.state("currentConsumer")).toEqual({});
      expect(wrapper.state("modalErrorMessage")).toEqual("");
    });
  });

  describe("isSameBudget", () => {
    it("should be true", () => {
      const wrapper = shallow(<Main />);
      const instance = wrapper.instance();

      expect(
        instance.isSameBudget({ budget_raw: 1, budget_untouched: 1 })
      ).toBe(true);
    });

    it("should be false", () => {
      const wrapper = shallow(<Main />);
      const instance = wrapper.instance();
      const consumer = { budget_raw: 1, budget_untouched: 2 };
      expect(instance.isSameBudget(consumer)).toBe(false);
    });
  });

  describe("isValidBudget", () => {
    it("should be true", () => {
      const wrapper = shallow(<Main />);
      const instance = wrapper.instance();
      const consumer = { budget_raw: 1 };
      expect(instance.isValidBudget(consumer)).toBe(true);
    });

    it("should be false", () => {
      const wrapper = shallow(<Main />);
      const instance = wrapper.instance();
      const consumer = { budget_raw: "abc" };
      expect(instance.isValidBudget(consumer)).toBe(false);
    });
  });

  describe("isMoreThanSpent", () => {
    it("should be true", () => {
      const wrapper = shallow(<Main />);
      const instance = wrapper.instance();
      const consumer = { budget_raw: 2, budget_spent: 1 };
      expect(instance.isMoreThanSpent(consumer)).toBe(true);
    });

    it("should be true", () => {
      const wrapper = shallow(<Main />);
      const instance = wrapper.instance();
      const consumer = { budget_raw: 1, budget_spent: 1 };
      expect(instance.isMoreThanSpent(consumer)).toBe(true);
    });

    it("should be false", () => {
      const wrapper = shallow(<Main />);
      const instance = wrapper.instance();
      const consumer = { budget_raw: 0, budget_spent: 1 };
      expect(instance.isMoreThanSpent(consumer)).toBe(false);
    });
  });

  describe("updateConsumerBudget", () => {
    it("should update one item within the consumers list", () => {
      const wrapper = shallow(<Main />);

      const currentConsumer = {
        ...mockedConsumer,
        budget: 111222333.44,
        budget_raw: 111222333.44
      };

      wrapper.setState({
        modalOpen: true,
        currentConsumer,
        modalErrorMessage: ""
      });

      const instance = wrapper.instance();
      instance.updateConsumerBudget(currentConsumer);

      expect(wrapper.state().consumers[0]).toEqual(currentConsumer);
    });
  });
});
