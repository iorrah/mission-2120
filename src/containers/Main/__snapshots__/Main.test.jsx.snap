// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should mount correctly 1`] = `ReactWrapper {}`;

exports[`Main should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main__container"
      >
        <div
          class="main__delimiter"
        >
          <div
            class="main__header"
          >
            <h1
              class="main__title"
            >
              Consumers
              <small
                class="main__counter"
              >
                (
                13
                )
              </small>
            </h1>
            <div
              class="main__actions"
            >
              <button
                class="main__button-light btn btn-light"
                type="button"
              >
                Filter
                <img
                  alt="Filter"
                  class="main__filter-icon"
                  src="img/icon-14.png"
                />
              </button>
              <button
                class="main__button-primary btn btn-primary"
                type="button"
              >
                Add Consumer
                <img
                  alt="Add"
                  class="main__plus-icon"
                  src="img/icon-13.png"
                />
              </button>
            </div>
          </div>
          <table
            class="main__table"
          >
            <thead>
              <tr
                class="main__table-header-row"
              >
                <th
                  class="main__table-header-checkbox"
                >
                  <input
                    class="main__checkbox"
                    type="checkbox"
                  />
                </th>
                <th
                  class="main__table-header-id"
                >
                  ID
                </th>
                <th
                  class="main__table-header-logo"
                >
                  COMPANY
                </th>
                <th
                  class="main__table-header-name"
                >
                  NAME
                </th>
                <th
                  class="main__table-header-budget"
                >
                  BUDGET
                </th>
                <th
                  class="main__table-header-budget-spent"
                >
                  BUDGET SPENT
                </th>
                <th
                  class="main__table-header-date"
                >
                  DATE OF FIRST PURCHASE
                </th>
                <th>
                   
                </th>
              </tr>
            </thead>
            <tbody
              class="main__table-body"
            >
              <tr
                class="main__table-row"
              >
                <td>
                  <input
                    class="main__checkbox"
                    type="checkbox"
                  />
                </td>
                <td>
                  1
                </td>
                <td>
                  <img
                    alt="Company Logo"
                    class="main__row-logo"
                    src="img/logo-1.png"
                  />
                </td>
                <td>
                  Starbucks
                </td>
                <td>
                  €
                  94.726,57
                </td>
                <td>
                  €
                  5.783,71
                </td>
                <td>
                  16/03/2119
                </td>
                <td>
                  <img
                    alt="Further actions"
                    class="main__row-action"
                    src="img/icon-8.png"
                  />
                </td>
              </tr>
              <tr
                class="main__table-row"
              >
                <td>
                  <input
                    class="main__checkbox"
                    type="checkbox"
                  />
                </td>
                <td>
                  2
                </td>
                <td>
                  <img
                    alt="Company Logo"
                    class="main__row-logo"
                    src="img/logo-2.png"
                  />
                </td>
                <td>
                  NBC
                </td>
                <td>
                  €
                  56.104,03
                </td>
                <td>
                  €
                  3.943,57
                </td>
                <td>
                  20/09/2119
                </td>
                <td>
                  <img
                    alt="Further actions"
                    class="main__row-action"
                    src="img/icon-8.png"
                  />
                </td>
              </tr>
              <tr
                class="main__table-row"
              >
                <td>
                  <input
                    class="main__checkbox"
                    type="checkbox"
                  />
                </td>
                <td>
                  3
                </td>
                <td>
                  <img
                    alt="Company Logo"
                    class="main__row-logo"
                    src="img/logo-3.png"
                  />
                </td>
                <td>
                  Dominos
                </td>
                <td>
                  €
                  17.573,66
                </td>
                <td>
                  €
                  6.852,01
                </td>
                <td>
                  21/01/2119
                </td>
                <td>
                  <img
                    alt="Further actions"
                    class="main__row-action"
                    src="img/icon-8.png"
                  />
                </td>
              </tr>
              <tr
                class="main__table-row"
              >
                <td>
                  <input
                    class="main__checkbox"
                    type="checkbox"
                  />
                </td>
                <td>
                  4
                </td>
                <td>
                  <img
                    alt="Company Logo"
                    class="main__row-logo"
                    src="img/logo-4.png"
                  />
                </td>
                <td>
                  Shell
                </td>
                <td>
                  €
                  94.624,31
                </td>
                <td>
                  €
                  1.049,93
                </td>
                <td>
                  30/12/2119
                </td>
                <td>
                  <img
                    alt="Further actions"
                    class="main__row-action"
                    src="img/icon-8.png"
                  />
                </td>
              </tr>
              <tr
                class="main__table-row"
              >
                <td>
                  <input
                    class="main__checkbox"
                    type="checkbox"
                  />
                </td>
                <td>
                  5
                </td>
                <td>
                  <img
                    alt="Company Logo"
                    class="main__row-logo"
                    src="img/logo-5.png"
                  />
                </td>
                <td>
                  Pepsi
                </td>
                <td>
                  €
                  46.105,36
                </td>
                <td>
                  €
                  2.934,57
                </td>
                <td>
                  14/07/2119
                </td>
                <td>
                  <img
                    alt="Further actions"
                    class="main__row-action"
                    src="img/icon-8.png"
                  />
                </td>
              </tr>
              <tr
                class="main__table-row"
              >
                <td>
                  <input
                    class="main__checkbox"
                    type="checkbox"
                  />
                </td>
                <td>
                  6
                </td>
                <td>
                  <img
                    alt="Company Logo"
                    class="main__row-logo"
                    src="img/logo-6.png"
                  />
                </td>
                <td>
                  Dove
                </td>
                <td>
                  €
                  89.240,46
                </td>
                <td>
                  €
                  5.823,50
                </td>
                <td>
                  07/10/2119
                </td>
                <td>
                  <img
                    alt="Further actions"
                    class="main__row-action"
                    src="img/icon-8.png"
                  />
                </td>
              </tr>
              <tr
                class="main__table-row"
              >
                <td>
                  <input
                    class="main__checkbox"
                    type="checkbox"
                  />
                </td>
                <td>
                  7
                </td>
                <td>
                  <img
                    alt="Company Logo"
                    class="main__row-logo"
                    src="img/logo-7.png"
                  />
                </td>
                <td>
                  Dell
                </td>
                <td>
                  €
                  30.194,16
                </td>
                <td>
                  €
                  9.494,40
                </td>
                <td>
                  07/01/2119
                </td>
                <td>
                  <img
                    alt="Further actions"
                    class="main__row-action"
                    src="img/icon-8.png"
                  />
                </td>
              </tr>
              <tr
                class="main__table-row"
              >
                <td>
                  <input
                    class="main__checkbox"
                    type="checkbox"
                  />
                </td>
                <td>
                  8
                </td>
                <td>
                  <img
                    alt="Company Logo"
                    class="main__row-logo"
                    src="img/logo-8.png"
                  />
                </td>
                <td>
                  Fanta
                </td>
                <td>
                  €
                  35.124,10
                </td>
                <td>
                  €
                  4.935,15
                </td>
                <td>
                  02/04/2119
                </td>
                <td>
                  <img
                    alt="Further actions"
                    class="main__row-action"
                    src="img/icon-8.png"
                  />
                </td>
              </tr>
              <tr
                class="main__table-row"
              >
                <td>
                  <input
                    class="main__checkbox"
                    type="checkbox"
                  />
                </td>
                <td>
                  9
                </td>
                <td>
                  <img
                    alt="Company Logo"
                    class="main__row-logo"
                    src="img/logo-9.png"
                  />
                </td>
                <td>
                  LG
                </td>
                <td>
                  €
                  64.948,15
                </td>
                <td>
                  €
                  4.158,50
                </td>
                <td>
                  19/03/2119
                </td>
                <td>
                  <img
                    alt="Further actions"
                    class="main__row-action"
                    src="img/icon-8.png"
                  />
                </td>
              </tr>
              <tr
                class="main__table-row"
              >
                <td>
                  <input
                    class="main__checkbox"
                    type="checkbox"
                  />
                </td>
                <td>
                  10
                </td>
                <td>
                  <img
                    alt="Company Logo"
                    class="main__row-logo"
                    src="img/logo-10.png"
                  />
                </td>
                <td>
                  Volkswagen
                </td>
                <td>
                  €
                  10.353,30
                </td>
                <td>
                  €
                  9.485,76
                </td>
                <td>
                  15/06/2119
                </td>
                <td>
                  <img
                    alt="Further actions"
                    class="main__row-action"
                    src="img/icon-8.png"
                  />
                </td>
              </tr>
              <tr
                class="main__table-row"
              >
                <td>
                  <input
                    class="main__checkbox"
                    type="checkbox"
                  />
                </td>
                <td>
                  11
                </td>
                <td>
                  <img
                    alt="Company Logo"
                    class="main__row-logo"
                    src="img/logo-11.png"
                  />
                </td>
                <td>
                  Adidas
                </td>
                <td>
                  €
                  12.833,88
                </td>
                <td>
                  €
                  5.772,56
                </td>
                <td>
                  03/10/2119
                </td>
                <td>
                  <img
                    alt="Further actions"
                    class="main__row-action"
                    src="img/icon-8.png"
                  />
                </td>
              </tr>
              <tr
                class="main__table-row"
              >
                <td>
                  <input
                    class="main__checkbox"
                    type="checkbox"
                  />
                </td>
                <td>
                  12
                </td>
                <td>
                  <img
                    alt="Company Logo"
                    class="main__row-logo"
                    src="img/logo-12.png"
                  />
                </td>
                <td>
                  Cisco
                </td>
                <td>
                  €
                  42.952,23
                </td>
                <td>
                  €
                  6.620,00
                </td>
                <td>
                  12/12/2119
                </td>
                <td>
                  <img
                    alt="Further actions"
                    class="main__row-action"
                    src="img/icon-8.png"
                  />
                </td>
              </tr>
              <tr
                class="main__table-row"
              >
                <td>
                  <input
                    class="main__checkbox"
                    type="checkbox"
                  />
                </td>
                <td>
                  13
                </td>
                <td>
                  <img
                    alt="Company Logo"
                    class="main__row-logo"
                    src="img/logo-13.png"
                  />
                </td>
                <td>
                  Airbnb
                </td>
                <td>
                  €
                  91.835,63
                </td>
                <td>
                  €
                  9.246,22
                </td>
                <td>
                  08/05/2119
                </td>
                <td>
                  <img
                    alt="Further actions"
                    class="main__row-action"
                    src="img/icon-8.png"
                  />
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main__container"
    >
      <div
        class="main__delimiter"
      >
        <div
          class="main__header"
        >
          <h1
            class="main__title"
          >
            Consumers
            <small
              class="main__counter"
            >
              (
              13
              )
            </small>
          </h1>
          <div
            class="main__actions"
          >
            <button
              class="main__button-light btn btn-light"
              type="button"
            >
              Filter
              <img
                alt="Filter"
                class="main__filter-icon"
                src="img/icon-14.png"
              />
            </button>
            <button
              class="main__button-primary btn btn-primary"
              type="button"
            >
              Add Consumer
              <img
                alt="Add"
                class="main__plus-icon"
                src="img/icon-13.png"
              />
            </button>
          </div>
        </div>
        <table
          class="main__table"
        >
          <thead>
            <tr
              class="main__table-header-row"
            >
              <th
                class="main__table-header-checkbox"
              >
                <input
                  class="main__checkbox"
                  type="checkbox"
                />
              </th>
              <th
                class="main__table-header-id"
              >
                ID
              </th>
              <th
                class="main__table-header-logo"
              >
                COMPANY
              </th>
              <th
                class="main__table-header-name"
              >
                NAME
              </th>
              <th
                class="main__table-header-budget"
              >
                BUDGET
              </th>
              <th
                class="main__table-header-budget-spent"
              >
                BUDGET SPENT
              </th>
              <th
                class="main__table-header-date"
              >
                DATE OF FIRST PURCHASE
              </th>
              <th>
                 
              </th>
            </tr>
          </thead>
          <tbody
            class="main__table-body"
          >
            <tr
              class="main__table-row"
            >
              <td>
                <input
                  class="main__checkbox"
                  type="checkbox"
                />
              </td>
              <td>
                1
              </td>
              <td>
                <img
                  alt="Company Logo"
                  class="main__row-logo"
                  src="img/logo-1.png"
                />
              </td>
              <td>
                Starbucks
              </td>
              <td>
                €
                94.726,57
              </td>
              <td>
                €
                5.783,71
              </td>
              <td>
                16/03/2119
              </td>
              <td>
                <img
                  alt="Further actions"
                  class="main__row-action"
                  src="img/icon-8.png"
                />
              </td>
            </tr>
            <tr
              class="main__table-row"
            >
              <td>
                <input
                  class="main__checkbox"
                  type="checkbox"
                />
              </td>
              <td>
                2
              </td>
              <td>
                <img
                  alt="Company Logo"
                  class="main__row-logo"
                  src="img/logo-2.png"
                />
              </td>
              <td>
                NBC
              </td>
              <td>
                €
                56.104,03
              </td>
              <td>
                €
                3.943,57
              </td>
              <td>
                20/09/2119
              </td>
              <td>
                <img
                  alt="Further actions"
                  class="main__row-action"
                  src="img/icon-8.png"
                />
              </td>
            </tr>
            <tr
              class="main__table-row"
            >
              <td>
                <input
                  class="main__checkbox"
                  type="checkbox"
                />
              </td>
              <td>
                3
              </td>
              <td>
                <img
                  alt="Company Logo"
                  class="main__row-logo"
                  src="img/logo-3.png"
                />
              </td>
              <td>
                Dominos
              </td>
              <td>
                €
                17.573,66
              </td>
              <td>
                €
                6.852,01
              </td>
              <td>
                21/01/2119
              </td>
              <td>
                <img
                  alt="Further actions"
                  class="main__row-action"
                  src="img/icon-8.png"
                />
              </td>
            </tr>
            <tr
              class="main__table-row"
            >
              <td>
                <input
                  class="main__checkbox"
                  type="checkbox"
                />
              </td>
              <td>
                4
              </td>
              <td>
                <img
                  alt="Company Logo"
                  class="main__row-logo"
                  src="img/logo-4.png"
                />
              </td>
              <td>
                Shell
              </td>
              <td>
                €
                94.624,31
              </td>
              <td>
                €
                1.049,93
              </td>
              <td>
                30/12/2119
              </td>
              <td>
                <img
                  alt="Further actions"
                  class="main__row-action"
                  src="img/icon-8.png"
                />
              </td>
            </tr>
            <tr
              class="main__table-row"
            >
              <td>
                <input
                  class="main__checkbox"
                  type="checkbox"
                />
              </td>
              <td>
                5
              </td>
              <td>
                <img
                  alt="Company Logo"
                  class="main__row-logo"
                  src="img/logo-5.png"
                />
              </td>
              <td>
                Pepsi
              </td>
              <td>
                €
                46.105,36
              </td>
              <td>
                €
                2.934,57
              </td>
              <td>
                14/07/2119
              </td>
              <td>
                <img
                  alt="Further actions"
                  class="main__row-action"
                  src="img/icon-8.png"
                />
              </td>
            </tr>
            <tr
              class="main__table-row"
            >
              <td>
                <input
                  class="main__checkbox"
                  type="checkbox"
                />
              </td>
              <td>
                6
              </td>
              <td>
                <img
                  alt="Company Logo"
                  class="main__row-logo"
                  src="img/logo-6.png"
                />
              </td>
              <td>
                Dove
              </td>
              <td>
                €
                89.240,46
              </td>
              <td>
                €
                5.823,50
              </td>
              <td>
                07/10/2119
              </td>
              <td>
                <img
                  alt="Further actions"
                  class="main__row-action"
                  src="img/icon-8.png"
                />
              </td>
            </tr>
            <tr
              class="main__table-row"
            >
              <td>
                <input
                  class="main__checkbox"
                  type="checkbox"
                />
              </td>
              <td>
                7
              </td>
              <td>
                <img
                  alt="Company Logo"
                  class="main__row-logo"
                  src="img/logo-7.png"
                />
              </td>
              <td>
                Dell
              </td>
              <td>
                €
                30.194,16
              </td>
              <td>
                €
                9.494,40
              </td>
              <td>
                07/01/2119
              </td>
              <td>
                <img
                  alt="Further actions"
                  class="main__row-action"
                  src="img/icon-8.png"
                />
              </td>
            </tr>
            <tr
              class="main__table-row"
            >
              <td>
                <input
                  class="main__checkbox"
                  type="checkbox"
                />
              </td>
              <td>
                8
              </td>
              <td>
                <img
                  alt="Company Logo"
                  class="main__row-logo"
                  src="img/logo-8.png"
                />
              </td>
              <td>
                Fanta
              </td>
              <td>
                €
                35.124,10
              </td>
              <td>
                €
                4.935,15
              </td>
              <td>
                02/04/2119
              </td>
              <td>
                <img
                  alt="Further actions"
                  class="main__row-action"
                  src="img/icon-8.png"
                />
              </td>
            </tr>
            <tr
              class="main__table-row"
            >
              <td>
                <input
                  class="main__checkbox"
                  type="checkbox"
                />
              </td>
              <td>
                9
              </td>
              <td>
                <img
                  alt="Company Logo"
                  class="main__row-logo"
                  src="img/logo-9.png"
                />
              </td>
              <td>
                LG
              </td>
              <td>
                €
                64.948,15
              </td>
              <td>
                €
                4.158,50
              </td>
              <td>
                19/03/2119
              </td>
              <td>
                <img
                  alt="Further actions"
                  class="main__row-action"
                  src="img/icon-8.png"
                />
              </td>
            </tr>
            <tr
              class="main__table-row"
            >
              <td>
                <input
                  class="main__checkbox"
                  type="checkbox"
                />
              </td>
              <td>
                10
              </td>
              <td>
                <img
                  alt="Company Logo"
                  class="main__row-logo"
                  src="img/logo-10.png"
                />
              </td>
              <td>
                Volkswagen
              </td>
              <td>
                €
                10.353,30
              </td>
              <td>
                €
                9.485,76
              </td>
              <td>
                15/06/2119
              </td>
              <td>
                <img
                  alt="Further actions"
                  class="main__row-action"
                  src="img/icon-8.png"
                />
              </td>
            </tr>
            <tr
              class="main__table-row"
            >
              <td>
                <input
                  class="main__checkbox"
                  type="checkbox"
                />
              </td>
              <td>
                11
              </td>
              <td>
                <img
                  alt="Company Logo"
                  class="main__row-logo"
                  src="img/logo-11.png"
                />
              </td>
              <td>
                Adidas
              </td>
              <td>
                €
                12.833,88
              </td>
              <td>
                €
                5.772,56
              </td>
              <td>
                03/10/2119
              </td>
              <td>
                <img
                  alt="Further actions"
                  class="main__row-action"
                  src="img/icon-8.png"
                />
              </td>
            </tr>
            <tr
              class="main__table-row"
            >
              <td>
                <input
                  class="main__checkbox"
                  type="checkbox"
                />
              </td>
              <td>
                12
              </td>
              <td>
                <img
                  alt="Company Logo"
                  class="main__row-logo"
                  src="img/logo-12.png"
                />
              </td>
              <td>
                Cisco
              </td>
              <td>
                €
                42.952,23
              </td>
              <td>
                €
                6.620,00
              </td>
              <td>
                12/12/2119
              </td>
              <td>
                <img
                  alt="Further actions"
                  class="main__row-action"
                  src="img/icon-8.png"
                />
              </td>
            </tr>
            <tr
              class="main__table-row"
            >
              <td>
                <input
                  class="main__checkbox"
                  type="checkbox"
                />
              </td>
              <td>
                13
              </td>
              <td>
                <img
                  alt="Company Logo"
                  class="main__row-logo"
                  src="img/logo-13.png"
                />
              </td>
              <td>
                Airbnb
              </td>
              <td>
                €
                91.835,63
              </td>
              <td>
                €
                9.246,22
              </td>
              <td>
                08/05/2119
              </td>
              <td>
                <img
                  alt="Further actions"
                  class="main__row-action"
                  src="img/icon-8.png"
                />
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Main should shallow correctly 1`] = `ShallowWrapper {}`;
